gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/offchainConfig/begin.ts

```diff
@@ -19,33 +19,68 @@ export default class BeginOffchainConfig extends SolanaCommand {
     this.require(!!this.flags.state, 'Please provide flags with "state"')
   }
 
-  execute = async () => {
+  makeRawTransaction = async (signer: PublicKey) => {
     const ocr2 = getContract(CONTRACT_LIST.OCR_2, '')
     const address = ocr2.programId.toString()
     const program = this.loadProgram(ocr2.idl, address)
 
     const state = new PublicKey(this.flags.state)
-    const owner = this.wallet.payer
     const version = new BN(2)
 
     await prompt(`Begin setting Offchain config version ${version.toString()}?`)
 
-    const tx = await program.rpc.beginOffchainConfig(version, {
-      accounts: {
-        state: state,
-        authority: owner.publicKey,
-      },
+    const data = program.coder.instruction.encode('begin_offchain_config', {
+      offchainConfigVersion: version,
     })
 
-    logger.success(`Begin set offchain config on tx ${tx}`)
+    const accounts: AccountMeta[] = [
+      {
+        pubkey: state,
+        isSigner: false,
+        isWritable: true,
+      },
+      {
+        pubkey: signer,
+        isSigner: true,
+        isWritable: false,
+      },
+    ]
+
+    const rawTx: RawTransaction = {
+      data,
+      accounts,
+      programId: ocr2.programId,
+    }
+
+    return [rawTx]
+  }
+
+  execute = async () => {
+    const rawTx = await this.makeRawTransaction(this.wallet.payer.publicKey)
+    const tx = rawTx.reduce(
+      (tx, meta) =>
+        tx.add(
+          new TransactionInstruction({
+            programId: meta.programId,
+            keys: meta.accounts,
+            data: meta.data,
+          }),
+        ),
+      new Transaction(),
+    )
+    logger.loading('Sending tx...')
+    logger.debug(tx)
+    const txhash = await this.provider.send(tx, [this.wallet.payer])
```

Comment:
Can we find a way to map the program errors that are thrown here to the respective messages, like the .rpc. calls do?

Currently we get back the hex error code, e.g. custom program error : 0xbc2(converting it into decimal makes it possible to find/search for the error code and see the message)

Reply:
Anchor `parseIdlErrors` library will be used